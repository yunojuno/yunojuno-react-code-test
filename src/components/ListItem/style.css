.listItem {
  width: 100%;

  display: grid;
  grid-template-rows: 2rem 1fr;
  grid-template-columns: 5rem 1fr;
  grid-column-gap: 1rem;
  grid-row-gap: 1rem;
  justify-items: flex-start;
  align-items: center;

  margin: 1rem 0;
  padding: 1rem;

  text-align: left;
  background-color: #47d7ac;
  color: white;
}

.listItem__avatar {
  grid-row: 1/3;
  grid-column: 1/2;

  justify-self: center;
  align-self: flex-start;

  border-radius: 100%;
  width: 5rem;
  height: 5rem;

  background-color: grey;
}

.listItem__details {
  font-weight: bold;
  align-self: flex-end;
}

.listItem__description {
  align-self: flex-start;
}

@media only screen and (max-width: 1000px) {
  .listItem {
    grid-template-rows: 5rem auto;
    grid-template-columns: 1fr;
    grid-column-gap: 0;
    grid-row-gap: 0.5rem;
  }
}
