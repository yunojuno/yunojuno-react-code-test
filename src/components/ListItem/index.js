import React, { Component } from 'react';
import "./style.css";

export default class ListItem extends Component {
  render() {
    const { name, category, description, avatar } = this.props;

    return (
      <li className="listItem">
        <img alt={name} src={`${process.env.PUBLIC_URL}/characters/${avatar}`} className="listItem__avatar" />
        <span className="listItem__details">{`${name} (${category})`}</span>
        <span className="listItem__description">{description}</span>
      </li>
    );
  }
}
