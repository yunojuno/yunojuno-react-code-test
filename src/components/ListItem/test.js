import React from "react";
import { shallow } from 'enzyme';

import characters from "../../fixtures/characters.json";

import ListItem from '../ListItem/';

const testCharacter = characters[0];
const list = shallow(<ListItem {...testCharacter} />);

it("renders correctly", () => {
  expect(list).toMatchSnapshot();
});

it('renders an image', () => {
  expect(list.find('img').length).toBe(1);
});

it('renders the correct avatar', () => {
  expect(list.find('img').prop('src')).toContain(testCharacter.avatar);
});

it('renders the correct details', () => {
  expect(list.find('span').at(0).text()).toContain(testCharacter.name);
  expect(list.find('span').at(0).text()).toContain(testCharacter.category);
});
