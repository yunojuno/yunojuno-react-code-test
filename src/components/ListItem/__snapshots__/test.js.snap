// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListItem
    avatar="frodo_baggins.jpg"
    category="hobbit"
    description="A young well-to-do hobbit. When he discovers that the magic ring left to him by his eccentric Uncle Bilbo is the One Ring, he reluctantly takes on the quest to destroy it. Often referred to as the Ringbearer."
    name="Frodo Baggins"
    significanceIndex={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Frodo Baggins"
          className="listItem__avatar"
          src="/characters/frodo_baggins.jpg"
/>,
        <span
          className="listItem__details"
>
          Frodo Baggins (hobbit)
</span>,
        <span
          className="listItem__description"
>
          A young well-to-do hobbit. When he discovers that the magic ring left to him by his eccentric Uncle Bilbo is the One Ring, he reluctantly takes on the quest to destroy it. Often referred to as the Ringbearer.
</span>,
      ],
      "className": "listItem",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Frodo Baggins",
          "className": "listItem__avatar",
          "src": "/characters/frodo_baggins.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Frodo Baggins (hobbit)",
          "className": "listItem__details",
        },
        "ref": null,
        "rendered": "Frodo Baggins (hobbit)",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "A young well-to-do hobbit. When he discovers that the magic ring left to him by his eccentric Uncle Bilbo is the One Ring, he reluctantly takes on the quest to destroy it. Often referred to as the Ringbearer.",
          "className": "listItem__description",
        },
        "ref": null,
        "rendered": "A young well-to-do hobbit. When he discovers that the magic ring left to him by his eccentric Uncle Bilbo is the One Ring, he reluctantly takes on the quest to destroy it. Often referred to as the Ringbearer.",
        "type": "span",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Frodo Baggins"
            className="listItem__avatar"
            src="/characters/frodo_baggins.jpg"
/>,
          <span
            className="listItem__details"
>
            Frodo Baggins (hobbit)
</span>,
          <span
            className="listItem__description"
>
            A young well-to-do hobbit. When he discovers that the magic ring left to him by his eccentric Uncle Bilbo is the One Ring, he reluctantly takes on the quest to destroy it. Often referred to as the Ringbearer.
</span>,
        ],
        "className": "listItem",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Frodo Baggins",
            "className": "listItem__avatar",
            "src": "/characters/frodo_baggins.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Frodo Baggins (hobbit)",
            "className": "listItem__details",
          },
          "ref": null,
          "rendered": "Frodo Baggins (hobbit)",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "A young well-to-do hobbit. When he discovers that the magic ring left to him by his eccentric Uncle Bilbo is the One Ring, he reluctantly takes on the quest to destroy it. Often referred to as the Ringbearer.",
            "className": "listItem__description",
          },
          "ref": null,
          "rendered": "A young well-to-do hobbit. When he discovers that the magic ring left to him by his eccentric Uncle Bilbo is the One Ring, he reluctantly takes on the quest to destroy it. Often referred to as the Ringbearer.",
          "type": "span",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
